### ISSUE TITLE
Does this need hoe as runtime dependency ?

### ISSUE THREAD

**Issue by grosser** [halostatue-archive/transaction-simple#1]
As far as I can see it does not, would be great if it could be removed (development dependency is fine!)

**Comment by halostatue** [comment]
On Wed, May 30, 2012 at 1:23 AM, Michael Grosser <
reply@reply.github.com

> wrote:
> 
> As far as I can see it does not, would be great if it could be removed
> (development dependency is fine!)
> 
> It doesn't. A patch to fix this (as a development dependency) would be
> most welcomed—be warned, however, that it will be some time before I can
> apply it because I leave for Europe this evening (~two weeks) and I haven't
> packed yet.

It probably does this because the last time I changed the library was
before development dependencies existed.

## -a

Austin Ziegler • austin@zieglers.ca • austin@halostatue.ca
http://www.halostatue.ca/ • http://twitter.com/halostatue

**Comment by grosser** [comment]
Are you ok with removing hoe and making this use bundler for gem management ?

**Comment by halostatue** [comment]
Not so much. Make an argument in favour, and I could change my mind. I see
bundler as good for app deployments, not so much for gem development. Hoe
is a _great_ development/deployment tool, as far as I'm concerned.

Whatever is left behind, I want to still be able to use things like "rake
publish".

-a

On Wed, May 30, 2012 at 10:58 AM, Michael Grosser <
reply@reply.github.com

> wrote:
> 
> Are you ok with removing hoe and making this use bundler for gem
> management ?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> 
> https://github.com/halostatue/transaction-simple/issues/1#issuecomment-6012078

## 

Austin Ziegler • austin@zieglers.ca • austin@halostatue.ca
http://www.halostatue.ca/ • http://twitter.com/halostatue

**Comment by grosser** [comment]
Its just rake version:bump:patch && rake release then :)
On May 30, 2012 8:21 AM, "Austin Ziegler" <
reply@reply.github.com>
wrote:

**Comment by grosser** [comment]
Hope you had a pleasant stay, time for some merge action :)

**Comment by halostatue** [comment]
I'm not yet convinced. Let me play with it some—I _really_ like what hoe provides a gem developer as a full suite, and I don't see that this is going to replace what I like. I've updated other gems to have hoe as a develop dependency and bring it up to date. I'm going to pull your changes into a branch and play with them before I make a decision.

**Comment by halostatue** [comment]
I will say that I absolutely _don't_ like the version:bump:patch bit. I don't want a separate file for versioning; I want verification for versioning as hoe's "rake release VERSION=x.y.z" requires.

**Comment by grosser** [comment]
Hmm you do not like blindly releasing a new version ?
I'm more in the other extreme, I don't care what version I release as long as it is a minor/patch bump.
I think it is very useful once you maintain lots of gems, no more need to look at anything, just merge pull request and bump:patch, done :)

Afaik the current best-practice is to have a gem/version.rb so people can check against the version of a dependency, maybe have a look at some of your favorite gems.

But if you want I'll get rid of version.rb and make rake release VERSION=x.y.z work.

**Comment by halostatue** [comment]
In the hour or so since I started looking at this, I already have the modern-hoe conversion done, plus properly renaming the files that don't look good to .rdoc so they _do_ look good and still create good documentation.

I understand that a lot of gem developers have version.rb files, but that doesn't seem like a best practice to me at all. I do have a number of gems that I maintain sporadically, but I don't want to simple merge pull requests—I have things that I expect from contributions.

BTW, it's much more than just the version:bump:patch thing—I _use_ debug_email, publish_docs, etc. on a regular basis to update the Rubyforge sites (and I just realized that I had _not_ generated such a site for transaction-simple, so…fixed, now).

…and released. Thanks for the push to get this taken care of.
